src/DataCite.jl

```diff
@@ -38,13 +40,13 @@ function data_fullname(::DataCite, mainpage)
 end
 
 function website(repo::DataCite, mainpage_url)
-    replace(mainpage_url, "https://api.datacite.org/works/", "https://doi.org/")
+    replace(mainpage_url, base_url(repo), "https://doi.org/")
 end
 
 function mainpage_url(repo::DataCite, dataname)
```

Comment:
For my own future reference,
this is infact different in functionality to
https://github.com/oxinabox/DataDepsGenerators.jl/blob/master/src/DataDepsGenerators.jl#L113-L122

This one takes anything that countains a DOI, e.g. some other URL, and gets the DOI from it.
Where are that one only works if the correct full URL, or just a DOI is passed.